ndcube/ndcube.py

```diff
@@ -629,6 +630,74 @@ def explode_along_axis(self, axis):
         # Creating a new NDCubeSequence with the result_cubes and common axis as axis
         return NDCubeSequence(result_cubes, meta=self.meta)
 
+    def reproject(self, target_wcs, shape_out=None, order='bilinear', output_array=None, return_footprint=False):
+        """
+        Reprojects an NDCube on another WCS object to upsample or downsample the pixel resolution.
+
+        Parameters
+        ----------
+        target_wcs : `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
+            The WCS object on which the NDCube is to be reprojected.
+
+        shape_out: `tuple`, optional
+            The shape of the output data. If not specified, `pixel_shape` attribute (if available)
+            from the low level API of the `target_wcs` is used.
+
+        order: `int or str`
+            The order of the interpolation. This can be any of: 'nearest-neighbour', 'bilinear',
+            'biquadratic', or 'bicubic'.
+
+        output_array: `None or numpy.ndarray`
+            An array in which to store the reprojected data. This can be any numpy array
+            including a memory map, which may be helpful when dealing with extremely large files.
+
+        return_footprint: `bool`
+            Whether to return the footprint in addition to the output array.
+
+        Returns
+        -------
+        resampled_cube : `ndcube.NDCube`
+            A new resultant NDCube object.
+
+        footprint: `numpy.ndarray`
+            Footprint of the input array in the output array. Values of 0 indicate no coverage or
+            valid values in the input image, while values of 1 indicate valid values.
+
+        Notes
+        -----
+        This method doesn't support handling of `mask`, `extra_coords`, and `uncertainty` yet.
+        However, `meta` and `global_coords` are retained.
+        """
+
+        if not utils.wcs.compare_wcs_physical_types(self.wcs, target_wcs):
+            raise Exception('Given target_wcs is not compatible with this NDCube.')
+
+        low_level_target_wcs = utils.wcs.get_low_level_wcs(target_wcs, 'target_wcs')
+
+        # If shape_out is not specified explicity, try to extract it from the low level WCS
+        if not shape_out:
+            if hasattr(low_level_target_wcs, 'pixel_shape') and low_level_target_wcs.pixel_shape:
```

Comment:
You need to do the `is not None` any time you are comparing to a thing which could be an iterable.
```suggestion
            if hasattr(low_level_target_wcs, 'pixel_shape') and low_level_target_wcs.pixel_shape is not None:
```

Reply:
Got it